be/src/agent/task_worker_pool.cpp

```diff
@@ -1963,4 +1745,239 @@ Status StorageMediumMigrateTaskPool::_check_migrate_request(const TStorageMedium
     return Status::OK();
 }
 
+CheckConsistencyTaskPool::CheckConsistencyTaskPool(ExecEnv* env, ThreadModel thread_model)
```

Comment:
It's better to reduce duplicated code that declares `XxxTaskPool` multiple times by invoking the template.
Here is an example, the implementation of each callback (template specialization) can be put into a separate file
![img_v2_668c21f7-9817-43d2-ace1-1516f44843cg](https://github.com/apache/doris/assets/6919662/abc6dfd8-2a3c-4896-b78b-19bc2bb92f8c)